packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.test.js

```diff
@@ -128,101 +134,346 @@ describe('overrides plugin', () => {
-            null,
-            {},
-            callback
-        )
+        const {resolver, callback} = setupResolverAndCallback()
         const overridesResolver = new OverridesResolverPlugin(options)
         overridesResolver.handleHook(testRequestContext, {}, callback, resolver)
 
         expect(callback).toHaveBeenCalled()
-        expect(resolver.ensureHook).not.toHaveBeenCalled()
         expect(resolver.doResolve).not.toHaveBeenCalled()
     })
 
     test('a file that requests from relative AND base template is able to get both', () => {
         const REQUEST_ONE_PATH = 'exists'
         const REQUEST_ONE_EXTENSION = '.jsx'
-        const testOneRequestContext = {
-            _ResolverCachePluginCacheMiss: true,
-            context: {
-                issuer: path.join('./', 'fake-file.js')
-            },
-            path: path.join('./', 'node_modules', EXTENDS_TARGET),
+        const testOneRequestContext = createRequestContextWith({
+            path: path.join('.', 'node_modules', EXTENDS_TARGET),
             request: `${EXTENDS_TARGET}/${REQUEST_ONE_PATH}`
-        }
+        })
 
-        let {resolver, callback} = setupResolverAndCallback(
-            null,
-            testOneRequestContext,
-            null,
-            {},
-            callback
-        )
+        let {resolver, callback} = setupResolverAndCallback()
         const overridesResolver = new OverridesResolverPlugin(options)
         overridesResolver.handleHook(testOneRequestContext, {}, callback, resolver)
 
         expect(callback).toHaveBeenCalled()
-        expect(resolver.ensureHook).toHaveBeenCalled()
         expect(resolver.doResolve).toHaveBeenCalledWith(
             null,
-            {
-                _ResolverCachePluginCacheMiss: true,
-                context: {
-                    issuer: path.join('./', 'fake-file.js')
-                },
+            createRequestContextWith({
                 path: path.join(REWRITE_DIR, REQUEST_ONE_PATH + REQUEST_ONE_EXTENSION),
-                request: `retail-react-app/exists`
-            },
+                request: `${EXTENDS_TARGET}/exists`
+            }),
             expect.anything(),
             expect.anything(),
             expect.anything()
         )
 
-        // TODO: this might be `..\` on Windows?
         const REQUEST_TWO_PATH = `./exists`
-        const testTwoRequestContext = {
-            _ResolverCachePluginCacheMiss: true,
-            context: {
-                issuer: path.join('./', 'fake-file.js')
-            },
-            path: path.join('./'),
+        const testTwoRequestContext = createRequestContextWith({
+            path: './',
             request: REQUEST_TWO_PATH
-        }
-        ;({resolver, callback} = setupResolverAndCallback(
-            null,
-            testTwoRequestContext,
-            null,
-            {},
-            callback
-        ))
+        })
+
+        ;({resolver, callback} = setupResolverAndCallback())
```

Comment:
this looks funny, what are we doing with the leading `;`?

did you mean 

```
const {resolver, callback} = setupResolverAndCallback()
```

Reply:
Yes, I agree about this looking funny. I tried to get rid of the leading `;` as well but according to my IDE it's needed.

In this test, we are destructuring `{resolver, callback} = setupResolverAndCallback()` twice. If the 2nd destructure is not wrapped in `()`, or if we use a const, this 2nd destructure causes errors.

But if we wrap the destructure around `()` without the leading `;` a different error is thrown. I'm not quite sure what causes it but the compiler does not like it if we don't have the leading `;`